### ISSUE TITLE
varnish_default_config fails on Centos7

### ISSUE THREAD

**Issue by dpetzel** [rackspace-cookbooks/varnish#95]
I first converged the cookbook "as-is", and then added:

```
varnish_default_config 'default' do
  listen_port 80
  storage 'malloc'
  malloc_size  "#{(node['memory']['total'].to_i * 0.75).to_i}K"
end
```

Converging the node then yields:

```
             ================================================================================
             Error executing action `configure` on resource 'varnish_default_config[default]'
             ================================================================================

             Chef::Exceptions::ResourceNotFound
             ----------------------------------
             resource template[/etc/default/varnish] is configured to notify resource execute[systemctl-daemon-reload] with action run, but execute[systemctl-daemon-reload] cannot be found in the resource collection. template[/etc/default/varnish] is defined in /tmp/kitchen/cache/cookbooks/varnish/libraries/varnish_default_config.rb:64:in `configure_varnish_service'

             Cookbook Trace:
             ---------------
             /tmp/kitchen/cache/cookbooks/varnish/libraries/varnish_default_config.rb:79:in `configure_varnish_service'
             /tmp/kitchen/cache/cookbooks/varnish/libraries/varnish_default_config.rb:55:in `action_configure'

             Resource Declaration:
             ---------------------
             # In /tmp/kitchen/cache/cookbooks/mine_cookbook/recipes/default.rb
```

This is running version `varnish (2.3.0)` on `Chef 12.5.1`

**Comment by martinb3** [comment]
Hi there -- this cookbook currently has recipes that are the "old way" and resources that are part of the library cookbook rewrite. Mixing them isn't going to work currently.

**Comment by RyanJarv** [comment]
I think this could happen in v2.3.0 if the varnish_install resource was not used. varnish_install has been removed in 3.x in favor of installing the package manually or using the varnish::configure recipe and the execute[systemctl-daemon-reload] has been moved to varnish_config, so this should be fixed in version 3.0.0.
